pkgs/by-name/hy/hyprcursor/fix-icon-dirs.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/libhyprcursor/hyprcursor.cpp b/libhyprcursor/hyprcursor.cpp
+index 304ab9f..1f7e95d 100644
+--- a/libhyprcursor/hyprcursor.cpp
++++ b/libhyprcursor/hyprcursor.cpp
+@@ -14,7 +14,7 @@
+ using namespace Hyprcursor;
+
+ // directories for lookup
+-constexpr const std::array<const char*, 1> systemThemeDirs = {"/usr/share/icons"};
++constexpr const std::array<const char*, 2> systemThemeDirs = {"/usr/share/icons", "/run/current-system/sw/share/icons"};
```

Comment:
I copy the patch form https://github.com/hyprwm/hyprcursor/blob/main/nix/dirs.patch
but I still get error:
```
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 18.02µs. Estimated average calc time: 0.04µs.
terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
  what():  filesystem error: status: Permission denied [/usr/share/icons]

```

Reply:
Also we should pay attention to the `XCURSOR_PATH` environment variable

Reply:
What if you change it to remove `/usr/share/icons` completely?

I agree I should do a more extensive review of the code and try to patch it properly.

Reply:
Works on hyprcursor 0.1.6

Reply:
Does the included patch work on 0.1.7?

Reply:
> Does the included patch work on 0.1.7?

Yes